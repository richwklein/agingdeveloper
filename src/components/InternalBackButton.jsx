import React from "react";
import ChevronLeftIcon from "@mui/icons-material/ChevronLeft";
import {Button} from "@mui/material";
import PropTypes from "prop-types";
import InternalLink from "./InternalLink";

/**
 * Renders a contained button as an anchor with the "View all {name}" template.
 *
 * @param {InternalBackButtonProps} props
 * @return {React.ReactElement} - The react component
 */
export const InternalBackButton = ({name, path, useViewText=true}) => {
  const text = (useViewText == true ) ? `view all ${name}` : name;

  return (
    <Button
      startIcon={<ChevronLeftIcon />}
      variant="contained"
      color="secondary"
      disableElevation
      component={InternalLink}
      to={path}
      sx={{
        mx: 0,
        my: 0,
        textTransform: "capitalize",
      }}
    >
      {text}
    </Button>
  );
};

/**
 * @typedef InternalBackButtonProps - The internal back button props
 * @property {string} name - The name of the view to put in the text.
 * @property {string} path - The path to the view.
 * @property {boolean} [useViewText=true] - If the text should include "view all".
 */
InternalBackButton.propTypes = {
  name: PropTypes.string.isRequired,
  path: PropTypes.string.isRequired,
  useViewText: PropTypes.bool,
};

export default InternalBackButton;
