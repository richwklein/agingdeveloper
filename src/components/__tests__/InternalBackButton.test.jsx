import React from "react";
import {cleanup, render} from "@testing-library/react";
import InternalBackButton from "../InternalBackButton";

describe("InternalBackButton", () => {
  afterEach(cleanup);

  const renderComponent = (useViewText=true) => render(
      <InternalBackButton name="authors" path="/author" useViewText={useViewText} />,
  );

  test("that it renders correctly.", () => {
    const {container} = renderComponent();
    expect(container.firstChild).toMatchSnapshot();
  });

  test("that it contains the text.", () => {
    const {container} = renderComponent();
    expect(container.firstChild).toContainHTML("view all authors");
  });

  test("that it can exclude view all.", () => {
    const {container} = renderComponent(false);
    expect(container.firstChild).not.toContainHTML("view all");
  });

  test("that the icons and href are.", () => {
    const {container, getByTestId} = renderComponent();
    expect(getByTestId("ChevronLeftIcon")).toBeInTheDocument();
    expect(container.firstChild).toContainHTML("href=\"/author\"");
  });
});
