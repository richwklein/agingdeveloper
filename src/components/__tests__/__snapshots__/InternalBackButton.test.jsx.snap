// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InternalBackButton that it renders correctly. 1`] = `
<a
  class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation css-1vgpm7k-MuiTypography-root-MuiLink-root-MuiButtonBase-root-MuiButton-root"
  data-link="internal"
  href="/author"
  tabindex="0"
>
  <span
    class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="ChevronLeftIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M15.41 7.41 14 6l-6 6 6 6 1.41-1.41L10.83 12z"
      />
    </svg>
  </span>
  view all authors
  <span
    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
  />
</a>
`;

exports[`InternalBackButton that it renders with main. 1`] = `
<a
  class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation css-1vgpm7k-MuiTypography-root-MuiLink-root-MuiButtonBase-root-MuiButton-root"
  data-link="internal"
  href="/author"
  tabindex="0"
>
  <span
    class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="ChevronLeftIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M15.41 7.41 14 6l-6 6 6 6 1.41-1.41L10.83 12z"
      />
    </svg>
  </span>
  view all authors
  <span
    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
  />
</a>
`;
